### PR TITLE
Adjust etcd formula to patch the python-etcd module

### PR DESCRIPTION
**Pull request by petrudumuta**
- added state for optionally patching the etcd_client python module, in order for it to be compatible with etcd > v 3.

### CODE REVIEW

**Comment by petrudumuta** [comment]
@herisanu can you please review this?

File: etcd/files/etcd_client_patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
+--- /usr/lib/python2.7/site-packages/etcd/client.py	2016-12-22 13:41:54.913585308 +0000
```
Comment (herisanu) [inline etcd/files/etcd_client_patch]: could you please name this file as etcd_client_ssl_tlsv12.patch

**Review by herisanu — CHANGES_REQUESTED** [review]
- could you please change the name of the patch file to etcd_client_ssl_tlsv12.patch
- could you please attach proof it works by showing a etcd sdb backed connection working over SSL

**Comment by herisanu** [comment]
The reason for this PR is that out-of-the-box the python-etcd client will not work with etcd v3 using SSL.
It throws an exception that TLS v1 is not supported and there is no option to use another protocol.

**Comment by herisanu** [comment]
Hmm @dpetre I see this got addressed here:
https://github.com/jplana/python-etcd/commit/80f936963773053c36ebdff5185147b5e127f94d

Makes me wonder if it would be better to intstall the python dependency from the git url directly

**Comment by petrudumuta** [comment]
@herisanu I looked into how we could install the `python-etcd` python module from the revision you mentioned (that fixes the TLS issue) and while pip allows you to specify a commit hash, I couldn't find this option for the `pip_state` state in Salt (maybe I'm missing out something)

**Comment by herisanu** [comment]
This works with the latest version in git (python-etcd). Closing this out.

**Pull request closed without merging**